### PR TITLE
Utility refactoring for shared tool logic

### PR DESCRIPTION
**Pull request by BenWilson2**
Refactoring utilities for shared functionality needs with additional integrations

### CODE REVIEW

File: src/unitycatalog/ai/utils/client_utils.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def validate_or_set_default_client(client: Optional[BaseFunctionClient] = None):
+    """
+    Validate or set the default client.
+    If a client is provided, it returns the client. If not, it attempts to retrieve
+    the default client using `get_uc_function_client()`. Raises a `ValueError` if no
+    client is available.
+    Args:
+        client (Optional[BaseFunctionClient], optional): The client to validate or set.
```
Comment (serena-ruan) [inline src/unitycatalog/ai/utils/client_utils.py]:
```suggestion
        client (Optional[BaseFunctionClient]): The client to validate or set.
```

File: src/unitycatalog/ai/utils/function_processing_utils.py [hunk 2]
```diff
@@ -249,38 +97,64 @@ def uc_type_json_to_pydantic_type(
...
+                while True:
+                    functions = client.list_functions(
+                        catalog=full_func_name.catalog_name,
+                        schema=full_func_name.schema_name,
+                        max_results=max_results,
+                        page_token=token,
+                    )
+                    for f in functions:
+                        if f.full_name not in tools_dict:
+                            tools_dict[f.full_name] = uc_function_to_tool_func(function_info=f)
```
Comment (serena-ruan) [inline src/unitycatalog/ai/utils/function_processing_utils.py]:
```suggestion
                            tools_dict[f.full_name] = uc_function_to_tool_func(function_info=f, client=client)
```
Found this bug yesterday 😆 When the user only passes the client to the Toolkit and never sets default uc client, we need to pass the client to this function

File: src/unitycatalog/ai/utils/function_processing_utils.py [hunk 3]
```diff
@@ -249,38 +97,64 @@ def uc_type_json_to_pydantic_type(
...
+                    token = functions.token
+                    if token is None:
+                        break
+            else:
+                tools_dict[name] = uc_function_to_tool_func(function_name=name)
```
Comment (serena-ruan) [inline src/unitycatalog/ai/utils/function_processing_utils.py]:
```suggestion
                tools_dict[name] = uc_function_to_tool_func(function_name=name, client=client)
```

**Review by serena-ruan — APPROVED** [review]
LGTM! Thanks for refactoring this! Just a small change needed as I found that bug yesterday

File: src/unitycatalog/ai/utils/function_processing_utils.py (hunk 2 above)
Comment (BenWilson2) [reply src/unitycatalog/ai/utils/function_processing_utils.py]: ohhhhh great find! I totally missed that when doing my integration testing too!

**Comment by serena-ruan** [comment]
All tests validation in local:
image
image

**Pull request merged**